### PR TITLE
Feature/product availibility endpoint

### PR DESCRIPTION
**Pull request by rhimmelbauer**
Notes:

- Simple endpoint to toggle Product Availability along with Offers with related product

### CODE REVIEW

File: develop/core/tests/test_products.py [hunk 1]
```diff
@@ -195,6 +195,40 @@ def test_products_update_status_code_fail_no_login(self):
...
+        for offer in Offer.objects.filter(products__in=[self.product]):
+            offer.available = False
+            offer.save()
+
+        response = self.client.post(uri, data={'available': True})
+        self.product.refresh_from_db()
+        
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(self.product.available)
+        self.assertTrue(next((offer.available for offer in Offer.objects.filter(products__in=[self.product]) if offer.available), False))
```
Comment (devo-wm) [inline develop/core/tests/test_products.py:213]: should this be all() and not next() @rhimmelbauer

Comment (devo-wm) [reply develop/core/tests/test_products.py:213]:
```python
self.assertTrue(all(offer.available for offer in Offer.objects.filter(products__in=[self.product])))
```

File: develop/core/tests/test_products.py [hunk 2]
```diff
@@ -195,6 +195,40 @@ def test_products_update_status_code_fail_no_login(self):
...
+        for offer in Offer.objects.filter(products__in=[self.product]):
+            offer.available = True
+            offer.save()
+
+        response = self.client.post(uri, data={'available': False})
+        self.product.refresh_from_db()
+        
+        self.assertEqual(response.status_code, 302)
+        self.assertFalse(self.product.available)
+        self.assertFalse(next((offer.available for offer in Offer.objects.filter(products__in=[self.product]) if not offer.available), True))
```
Comment (devo-wm) [inline develop/core/tests/test_products.py:230]: See comment on here https://github.com/renderbox/django-vendor/pull/153/files#r616903351

File: vendor/views/vendor_admin.py [hunk 3]
```diff
@@ -331,3 +330,20 @@ def get(self, request, *args, **kwargs):
 
         messages.info(request, _("Offer Added To Customer Profile"))
         return redirect(reverse('vendor_admin:manager-profile', kwargs={'uuid': customer_profile.uuid}))
+
+
+class ProductAvailabilityToggleView(LoginRequiredMixin, View):
+    success_url = reverse_lazy('vendor_admin:manager-product-list')
+
+    def post(self, request, *args, **kwargs):
```
Comment (devo-wm) [inline vendor/views/vendor_admin.py:338]: Definitely looks like this'll work but wonder why we aren't just using an UpdateView?

File: vendor/views/vendor_admin.py [hunk 4]
```diff
@@ -331,3 +330,20 @@ def get(self, request, *args, **kwargs):
...
+        product = get_object_or_404(Product, uuid=self.kwargs.get('uuid'))
+
+        product.available = request.POST.get('available', False)
+        product.save()
+
+        for offer in Offer.objects.filter(products__in=[product]):
+            offer.available = product.available
+            offer.save()
+
+        messages.info(request, _("Product availability Changed"))
```
Comment (devo-wm) [inline vendor/views/vendor_admin.py:348]: SuccessMessageMixin?

**Review by devo-wm — APPROVED** [review]

File: vendor/views/vendor_admin.py (hunk 3 above)
Comment (rhimmelbauer) [reply vendor/views/vendor_admin.py:338]: Don't know did not really think about it. Was treating it as a simple endpoint not like a View

**Pull request merged**